lms/djangoapps/instructor_task/tests/test_tasks_helper.py

```diff
@@ -1670,7 +1670,7 @@ def test_certificate_generation_for_students(self):
             'failed': 3,
             'skipped': 2
         }
-        with self.assertNumQueries(175):
+        with self.assertNumQueries(191):
```

Comment:
Yeah - do note that it used to be 214 before the grades-refactor work landed (8/3/16).

The persisted Course Grades work should help with this path as well.

Also, we have not yet spent any time in optimizing for all the use cases where bulk-operations are needed across all students in a course.  Cert generation is one of them.
For example, in that case, we should be able to get the collected block structure once for all the students.